### PR TITLE
[Azure Search] Swagger specs and examples for data plane GA version 2…

### PR DESCRIPTION
**Pull request by bitsforthought**
…017-11-11

This updates the Swagger specs for the new GA version of the Azure Search data
plane. These new specs are exact copies of the 2016-09-01-Preview specs, with
the api-version changed. Same for the examples.

------------

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by brjohnstmsft** [comment]
@jianghaolu -- Can you please look at this today?

File: specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+      "name": "searchServiceName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-skip-url-encoding": true,
+      "description": "The name of the Azure Search service.",
+      "x-ms-parameter-location": "client"
+    },
+    "SearchDnsSuffixParameter": {
+      "name": "searchDnsSuffix",
```
Comment (jianghaolu) [inline specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json]: This required parameter is not provided in any example.

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/searchservice.json [hunk 2]
```diff
@@ -0,0 +1,4170 @@
...
+        }
+      ],
+      "properties": {
+        "preserveOriginal": {
+          "type": "boolean",
+          "default": false,
+          "description": "A value indicating whether the original token will be kept. Default is false."
+        }
+      },
+      "description": "Converts alphabetic, numeric, and symbolic Unicode characters which are not in the first 127 ASCII characters (the \"Basic Latin\" Unicode block) into their ASCII equivalents, if such equivalents exist. T
```
Comment (jianghaolu) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/searchservice.json]: Please add this required parameter to the examples.

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceCreateDataSource.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                "name": "myDocDbCollectionId",
+                "query": "SELECT c.id, c.userId, tag, c._ts FROM c JOIN tag IN c.tags WHERE c._ts >= @HighWaterMark ORDER BY c._ts"
+            },
+            "dataChangeDetectionPolicy": {
+                "@odata.type": "#Microsoft.Azure.Search.HighWaterMarkChangeDetectionPolicy",
+                "highWaterMarkColumnName": "_ts"
+            },
+            "dataDeletionDetectionPolicy": {
+                "@odata.type": "#Microsoft.Azure.Search.SoftDeleteColumnDeletionDetectionPolicy",
+                "softDeleteColumnName": "isDeleted",
```
Comment (jianghaolu) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceCreateDataSource.json]: These 2 properties are not defined in the type definition of "DataDeletionDetectionPolicy".

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceListIndexers.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        },
+                        "fieldMappings": [],
+                        "disabled": false  
+                    },
+                    {
+                        "name": "myotherindexer",  
+                        "description": "another cool indexer",  
+                        "dataSourceName": "myblobdatasource",  
+                        "targetIndexName": "orders",  
+                        "schedule": { },
```
Comment (jianghaolu) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceListIndexers.json]: "interval" is required here.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

**Comment by jianghaolu** [comment]
The GitHub review link is broken - The major valid error the tool is reporting is missing required parameters from the global "parameters" section, like "Prefer", or missing required parameters on the parameterized host, like "searchDnsSuffix".

File: specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json (hunk 1 above)
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json]: Do downstream consumers of these examples do anything useful with x-ms-parameterized-host parameters like this one? For example, when we start generating REST API documentation from Swagger (we aren't yet for our data plane, for historical reasons), will the examples in the docs put this parameter and `searchServiceName` in the host name part of the request URL where it should be?

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceCreateDataSource.json (hunk 3 above)
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceCreateDataSource.json]: This is a polymorphic type. Note the use of the `discriminator` property in `DataDeletionDetectionPolicy`. The value of `@odata.type` indicates that this object is of type `SoftDeleteColumnDeletionPolicy`. If you look at the definition of `SoftDeleteColumnDeletionPolicy` in the Swagger, you'll see the `softDeleteColumnName` and `softDeleteMarkerValue` properties are defined. Also note the use of `x-ms-discriminator-value` and `allOf` in `SoftDeleteColumnDeletionPolicy`.

Please ensure your validation tools understand polymorphism, as it is a crucial feature for modeling data plane APIs using AutoRest.

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceListIndexers.json (hunk 4 above)
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceListIndexers.json]: I think the example is meant to show that the schedule itself is not required. I'll remove it.

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/searchservice.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/searchservice.json]: I can't tell from the context -- Which parameter are you referring to?

Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/searchservice.json]: I can't tell from the context -- Which parameter are you referring to?

File: specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json (hunk 1 above)
Comment (amarzavery) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json]: The doc team is using the x-ms-examples for generating examples in the docs. I think they are doing that. [Here](https://docs.microsoft.com/en-us/rest/api/storagerp/srp_json_create_storage_account) is an example for the Storage Management API. Please contact the doc team for feature requests or any feedback.

Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11/searchindex.json]: @amarzavery Do you happen to know if [these KeyVault docs](https://docs.microsoft.com/rest/api/keyvault/createkey/createkey) are auto-generated as well? This seems to be an example of what we're hoping for.

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**